zk/client.go

```diff
@@ -33,16 +33,21 @@ func init() {
 // in ZooKeeper
 type ResourceName string
 
+// Prepend creates a new ResourceName with prefix prepended to the former ResourceName.
+func (r ResourceName) Prepend(prefix string) ResourceName {
+	return ResourceName(prefix + string(r))
```

Comment:
should use [`filepath.Join`](https://golang.org/pkg/path/filepath/#Join) in order to avoid double slashes if `ZookeeperRoot` has a trailing slash.